server/pipeline/cache.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const crypto = require('crypto')
+const tempy = require('tempy')
+const fs = require('fs')
+const path = require('path')
+const mkdir = require('make-dir')
+
+class Cache {
+	constructor({ filepath, contents }) {
+		this.filepath = filepath
+		this.data = contents
+
+		this.memcache = new Map()
+
+		if (!process.env.DOCKER) {
+			this.cacheDir = tempy.directory()
+		} else {
+			this.cacheDir = mkdir.sync('/tmp/hybsearch')
+		}
+
+		this.hashKey = this.hashKey.bind(this)
+		this.get = this.get.bind(this)
+		this.set = this.set.bind(this)
+		this._dataHash = this._hash(this.data)
+
+		this.set('source', { filepath: this.filepath, contents: this.data })
+	}
+
+	_hash(data) {
+		let hash = crypto.createHash('sha256')
+		hash.update(data)
+		return hash.digest('hex')
```

Comment:
I suppose since it's a digest it doesn't matter, but I'm partial to `base64`—no aversion to this, though.

Reply:
I don't even know what options I can give it. The docs only ever showed "hex".